lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -92,76 +94,86 @@ sub detect_asset_keys ($vars) {
-            log_error("Failed to download $asset_uri", channels => 'autoinst');
-            # assume that if we have a full path, that's what we should use
-            $vars->{$this_asset} = $asset_uri if -e $asset_uri;
-            # don't kill the job if the asset is not found
-            # TODO: This seems to leave the job stuck in some cases (observed in production on openqaworker3).
-            next;
-        }
-        if (!$asset) {
-            $error = "Failed to download $asset_uri to " . $cache_client->asset_path($webui_host, $asset_uri);
-            return {error => $error, category => WORKER_EC_ASSET_FAILURE} if $msg =~ qr/4\d\d /;
-            # note: This check has no effect if the download was performed by
-            # an already enqueued Minion job or if the pruning happened within
-            # a completely different asset download.
-            $error .= '. Asset was pruned immediately after download (poo#71827), please retrigger'
-              if $msg =~ /Purging.*$asset_uri.*because we need space for new assets/;
-            log_error($error, channels => 'autoinst');
-            return {error => $error};
-        }
-        my $link_result = _link_asset($asset, $pooldir);
-        $vars->{$this_asset}
-          = $vars->{ABSOLUTE_TEST_CONFIG_PATHS} ? $link_result->{absolute_path} : $link_result->{basename};
+    my $minion_id = $asset_request->minion_id;
+    log_info("Downloading $asset_uri, request #$minion_id sent to Cache Service", channels => 'autoinst');
+    return _poll_cache_service(
+        $job,
+        $cache_client,
+        $asset_request,
+        CACHE_SERVICE_POLL_DELAY,
+        sub ($error, $status) {
+            $error
+              = _handle_asset_processed($cache_client, $assets_to_cache, $asset_uri, $status, $vars, $webui_host,
+                $pooldir)
+              unless $error;
+            return $callback->($error) if $error;
+            return cache_assets($cache_client, $job, $vars, $assets_to_cache, $assetkeys, $webui_host, $pooldir,
+                $callback);
+        });
+}
+
+sub _handle_asset_processed ($cache_client, $this_asset, $asset_uri, $status, $vars, $webui_host, $pooldir) {
+    my $msg = "Download of $asset_uri processed";
+    if (my $output = $status->output) { $msg .= ":\n$output" }
+    log_info($msg, channels => 'autoinst');
+
+    my $asset
+      = $cache_client->asset_exists($webui_host, $asset_uri)
+      ? $cache_client->asset_path($webui_host, $asset_uri)
+      : undef;
+    if ($this_asset eq 'UEFI_PFLASH_VARS' && !defined $asset) {
+        log_error("Failed to download $asset_uri", channels => 'autoinst');
+        # assume that if we have a full path, that's what we should use
+        $vars->{$this_asset} = $asset_uri if -e $asset_uri;
+        # don't kill the job if the asset is not found
+        # TODO: This seems to leave the job stuck in some cases (observed in production on openqaworker3).
```

Comment:
I realize you didn't introduce this comment, but I feel like we should have a ticket instead of the comment. It's clearly not useful in getting somebody to fix it.

Reply:
Yes, this should have a ticket. However, for the sake of this change which is only supposed to turn synchronous functions into asynchronous ones it is out of scope (and I don't like to change multiple unrelated things in one PR).

I'm also afraid that it isn't easily possible to file a workable ticket at this point about it. So likely we can just remove the TODO completely. Without any clues why this would lead to stuck jobs it isn't very useful.

Reply:
Dunno if that single line is worth a PR. I just assume it'll stay in there and never get fixed.

Reply:
Please don't add TODO, same as disabled code as likely nobody else but you will understand this. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit, a ticket or just delete